x-pack/plugins/enterprise_search/public/applications/shared/licensing/license_context.test.tsx

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useContext } from 'react';
+
+import { mountWithKibanaContext } from '../../__mocks__';
+import { LicenseContext, ILicenseContext } from './';
+
+describe('LicenseProvider', () => {
+  const MockComponent: React.FC<> = () => {
+    const { license } = useContext(LicenseContext) as ILicenseContext;
+    return <div className="license-test">{license.type}</div>;
+  };
+
+  it('renders children', () => {
+    const wrapper = mountWithKibanaContext(
+      <LicenseContext.Provider value={{ license: { type: 'basic' } }}>
+        <MockComponent />
+      </LicenseContext.Provider>
+    );
+
+    expect(wrapper.find('.license-test')).toHaveLength(1);
+    expect(wrapper.text()).toEqual('basic');
```

Comment:
I would have liked to potentially add a test that mocks the observable changing data and the basic text changing to 'trial' or some such, but I'm not sure how to do so. Open to ideas if anyone has them!

Reply:
No ideas off of the top of my head.